File: testdata/bench/javascript/BenchmarkTest00003.js
// Vulnerable: SQL injection via template literal
const express = require('express');
const db = require('./db');
const router = express.Router();
router.post('/test003', (req, res) => {
    const name = req.body.name;
    const sql = `SELECT * FROM products WHERE name = '${name}'`;
    db.query(sql, (err, results) => { res.json(results); });
});
module.exports = router;

File: testdata/bench/kotlin/BenchmarkTest00003.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via string template with call.receive
suspend fun handler00003(call: ApplicationCall) {
    val body = call.receive<Map<String, String>>()
    val search = body["search"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    conn.createStatement().executeQuery("SELECT * FROM products WHERE name LIKE '%${search}%'")
    call.respondText("OK")
}
